main.cpp

```diff
@@ -322,11 +390,61 @@ int uthread_terminate(int tid) {
     } else if((it = find(waiting_list.begin(), waiting_list.end(), tid)) != waiting_list.end()) {
 	waiting_list.erase(it);
     } else {
-	return -1;
+	enable_interrupts();
+	thread_complete();  // If tid is the running thread
+	return 0;
```

Comment:
This is unrelated to interrupts, but I think we need to handle the case of where a thread tries to terminate itself. In that case we can just call the thread_complete() function.

Reply:
I think that should be an explicit `if(id == current_thread_id)`, else you could call on a **suspended** thread - That also needs to be handled here! - and shut yourself down.